Marketplace/AI_Nav/AI_Navigation.js

```diff
@@ -0,0 +1,352 @@
+                next = 0;
+                --numBeacons;
+                print("new array is length " + beacons.length);
+                //if there is only one beacon left then clear everything (must have at least two for app)
+                if (beacons.length == 1) {
+                    move = false;
+                    next = 0;
+                    numBeacons = 0;
+                    Entities.deleteEntity(beacons[0]);
+                    beacons = [];
+                    Entities.deleteEntity(model);
+                    model = undefined;
+                }
+            }
+        }
+
+        //makes the model move from beacon to beacon
+        if (move) {
+            //make object go to first beacon placed if its currently at the last beacon
+            if ((next) == numBeacons)
+            {
+                next = 0;
+            //will be else if distance is less than some number I will change target to next beacon
+            } else if (dist < .25) {
+                Script.update.disconnect(MakeMove);
+                var modelProperties = {
+                    velocity: {
+                        x: 0,
+                        y: 0,
+                        z: 0
+                    },
+                    animation: {
+                        url: idleAnim
+                    }
+                };
+                Entities.editEntity(model, modelProperties);
+                next++;
+                Script.setTimeout(Wait, (idleTime * 1000));
+            }
+        }
+    }
+
+    function Wait() {
+        var modelProperties = {
+                animation: {
+                    url: movingAnim
+                }
+            };
+        Entities.editEntity(model, modelProperties);
+        Script.update.connect(MakeMove);
+    }
+
+    function getVelocity(dx, dy, dz, dist, rot) {
+        var vectorProperties = {
+            velocity: {
+                x: (dx/dist) * speed,
+                y: (dy/dist) * speed,
+                z: (dz/dist) * speed
+            },
+            rotation: rot
+        };
+
+        return vectorProperties;
+    }
+
+    function getRotation(pointA, pointB, raise) {
+        // suppose we have a laser turret that we want to point at a
+        // known target location
+        pointA.y = raise;
+        var eye = pointB;
+        var center = pointA;
+        var up = Vec3.UP; // world-frame's up
+
+        // Quat.lookAt() will compute the orientation we want (MULTIPLIED TO MAKE LOOK BEHIND SINCE THATS HOW MODELS IMPORT)
+        var rot = Quat.multiply(Quat.lookAt(eye, center, up), {w: 0, x: 0, y: 1, z: 0});
+        return rot;
+    }
+
+    function getDistance(dx, dy, dz) {
+        //get distance between model and beacon
+        var dist = Math.sqrt(Math.pow(dx, 2) + Math.pow(dy, 2) + Math.pow(dz, 2));
+        return dist;
+    }
+
+    // Helper function that gives us a position right in front of the user 
+    function getPositionToCreateEntity() {

```

Comment:
Indentation should be 4 spaces, not 2

Reply:
Little confused on this one. Where is it two spaces?

Reply:
Nevermind I see what you mean now